src/itehandler/IteToSwitch.h

```diff
@@ -0,0 +1,177 @@
+//
+// Created by prova on 04.08.20.
+//
+
+#ifndef OPENSMT_ITETOSWITCH_H
+#define OPENSMT_ITETOSWITCH_H
+
+#include "Logic.h"
+
+
+namespace ite {
+    class timer {
+        time_t start;
+        const std::string name;
+    public:
+        explicit timer(std::string&& name) : start(time(nullptr)), name(name) {}
+        ~timer() { std::cout << name << ": " << time(nullptr) - start << std::endl; }
+    };
+
+    struct NodeRef {
+        uint32_t x;
+        inline friend bool operator== (NodeRef a1, NodeRef a2)   { return a1.x == a2.x; }
+        inline friend bool operator!= (NodeRef a1, NodeRef a2)   { return a1.x != a2.x; }
+        inline friend bool operator< (NodeRef a1, NodeRef a2)    { return a1.x < a2.x;  }
+    };
+    struct NodeRef_lt {
+        bool operator() (const NodeRef &n1, const NodeRef &n2) const { return n1 < n2; }
+    };
```

Comment:
I think the struct NodeRef_lt is not necessary, it is doing the same thing as operator< of NodeRef.